### ISSUE TITLE
False positive when appending a dict to a list of TypedDict unions

### ISSUE THREAD

**Issue by kkpattern** [astral-sh/ty#2216]
### Summary

The following code triggers a false positive invalid-argument-type error when appending a dictionary literal to a list of TypedDict unions:

```python
from typing import Literal, TypedDict


class Foo(TypedDict):
    role: Literal["assistant"]
    content: str

class Bar(TypedDict):
    role: Literal["user"]
    content: str


foo_list: list[Foo | Bar] = []
foo_list.append({"role": "assistant", "content": "hello world"})

# Argument to bound method `append` is incorrect: Expected `Foo | Bar`, found `dict[Unknown | str, Unknown | str]` (invalid-argument-type) [Ln 14, Col 17]
```

Both mypy and pyright pass without issues, but this tool reports an incorrect argument type.

Also the share button in the ty playground seems broken, I got the following error:

```
index-CzSy9k3J.js:54 Failed to share playground TypeError: Failed to fetch
    at H0 (index-CzSy9k3J.js:64:35618)
    at L0 (index-CzSy9k3J.js:64:35791)
    at index-CzSy9k3J.js:98:7335
    at onClick (index-CzSy9k3J.js:54:4234)
    at Zh (index-CzSy9k3J.js:49:117237)
    at index-CzSy9k3J.js:49:122342
    at Pc (index-CzSy9k3J.js:49:9040)
    at zu (index-CzSy9k3J.js:49:118471)
    at Gu (index-CzSy9k3J.js:50:26674)
    at Xb (index-CzSy9k3J.js:50:26496)
```

- [pyright playground](https://pyright-play.net/?code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAySMApiAIYA2ANFACqIkAmAIkgMYwCwAUL7%2B0rkAzsKgAxMGAAUDBMzacAlAC5eUDVHCUSKwsTJUA2gCIRwpMJjkUMEwF11m9mFslbeqyH49B5qABC5CCyjKwcMKpOGtq6%2BqQUlKYArsJkDtFQLm4eUF4%2BvMBSAPqUljB6ZVZGkmBQAD6BwfZQALxQRo48RWCl5QB05AjyKEzSAN4msSZ6ZqLlNna0JtmktjNQJgAWJJSUdQDuuJRMJgC%2BSkA)
- [mypy playground](https://mypy-play.net/?mypy=latest&python=3.12&gist=ada118463676e35110c6f35539168520)

### Version

ty 0.0.6

**Comment by charliermarsh** [comment]
(Can you share the browser you're on? The share button works fine for me -- just tested it.)

**Comment by kkpattern** [comment]
Windows 11
Chrome 143.0.7499.146.

I noticed there is another error log in the console:

```
Failed to load resource: net::ERR_CONNECTION_TIMED_OUT    api.astral-1ad.workers.dev/:1
```

It looks like a local network issue on my end, but the rest of the playground works fine.

BTW, Happy holidays!

**Comment by charliermarsh** [comment]
> BTW, Happy holidays!

You too!

**Comment by ibraheemdev** [comment]
It looks like the issue here is that the list element is a union of `TypedDict`s, which we currently don't support.
